tests/linearizability/traffic.go

```diff
@@ -17,39 +17,57 @@ package linearizability
 import (
 	"context"
 	"fmt"
+	"math"
 	"math/rand"
 	"time"
 
+	"go.uber.org/zap"
 	"golang.org/x/time/rate"
 
 	"go.etcd.io/etcd/api/v3/mvccpb"
+	"go.etcd.io/etcd/tests/v3/framework/config"
+	"go.etcd.io/etcd/tests/v3/framework/interfaces"
 	"go.etcd.io/etcd/tests/v3/linearizability/identity"
 	"go.etcd.io/etcd/tests/v3/linearizability/model"
 )
 
 var (
-	DefaultLeaseTTL int64   = 7200
-	RequestTimeout          = 40 * time.Millisecond
-	DefaultTraffic  Traffic = readWriteSingleKey{keyCount: 4, leaseTTL: DefaultLeaseTTL, writes: []opChance{{operation: model.Put, chance: 50}, {operation: model.Delete, chance: 10}, {operation: model.PutWithLease, chance: 10}, {operation: model.LeaseRevoke, chance: 10}, {operation: model.Txn, chance: 20}}}
+	startKey                       = "0"
+	endKey                         = fmt.Sprintf("%d", math.MaxInt)
+	DefaultLeaseTTL        int64   = 7200
+	RequestTimeout                 = 40 * time.Millisecond
+	DefaultTraffic         Traffic = readWriteSingleKey{keyCount: 4, leaseTTL: DefaultLeaseTTL, writes: []opChance{{operation: model.Put, chance: 50}, {operation: model.Delete, chance: 10}, {operation: model.PutWithLease, chance: 10}, {operation: model.LeaseRevoke, chance: 10}, {operation: model.Txn, chance: 20}}}
+	DefaultTrafficWithAuth Traffic = readWriteSingleKey{authEnabled: true, keyCount: 4, leaseTTL: DefaultLeaseTTL, writes: []opChance{{operation: model.Put, chance: 50}, {operation: model.Delete, chance: 10}, {operation: model.PutWithLease, chance: 10}, {operation: model.LeaseRevoke, chance: 10}, {operation: model.Txn, chance: 20}}}
 )
 
 type Traffic interface {
+	PreRun(ctx context.Context, c interfaces.Client, lg *zap.Logger) error
 	Run(ctx context.Context, clientId int, c *recordingClient, limiter *rate.Limiter, ids identity.Provider, lm identity.LeaseIdStorage)
+	AuthEnabled() bool
 }
 
 type readWriteSingleKey struct {
 	keyCount int
 	writes   []opChance
 	leaseTTL int64
+
+	authEnabled bool
 }
 
 type opChance struct {
 	operation model.OperationType
 	chance    int
 }
 
-func (t readWriteSingleKey) Run(ctx context.Context, clientId int, c *recordingClient, limiter *rate.Limiter, ids identity.Provider, lm identity.LeaseIdStorage) {
+func (t readWriteSingleKey) PreRun(ctx context.Context, c interfaces.Client, lg *zap.Logger) error {
+	if t.AuthEnabled() {
+		lg.Info("set up auth")
+		return setupAuth(ctx, c)
```

Comment:
Authorization setup should be done at cluster setup, not at this point.

Reply:
Make sense.